diff --git a/meshmc/CMakeLists.txt b/meshmc/CMakeLists.txt
index 39aaea51a8..48400b20b0 100644
--- a/meshmc/CMakeLists.txt
+++ b/meshmc/CMakeLists.txt
@@ -403,7 +403,7 @@ option(NBT_USE_ZLIB "Build NBT library with zlib support" OFF)
 option(NBT_BUILD_TESTS "Build NBT library tests" OFF) #FIXME: fix unit tests.
 set(NBT_NAME MeshMC_nbt++)
 set(NBT_DEST_DIR ${LIBRARY_DEST_DIR})
-add_subdirectory(${CMAKE_SOURCE_DIR}/../libnbtplusplus libnbtplusplus)
+add_subdirectory(libraries/libnbtplusplus)
 
 add_subdirectory(libraries/ganalytics) # google analytics library
 add_subdirectory(libraries/systeminfo) # system information library
diff --git a/meshmc/REUSE.toml b/meshmc/REUSE.toml
index 7583e77738..f9bf07171a 100644
--- a/meshmc/REUSE.toml
+++ b/meshmc/REUSE.toml
@@ -66,3 +66,8 @@ SPDX-FileCopyrightText = "NONE"
 path = ["CMakePresets.json"]
 SPDX-License-Identifier = "GPL-3.0-or-later"
 SPDX-FileCopyrightText = "2026 Project Tick"
+
+[[annotations]]
+path = ["libraries/libnbtplusplus/**"]
+SPDX-License-Identifier = "LGPL-3.0-or-later"
+SPDX-FileCopyrightText = "libnbtplusplus Contributors & ljfa-ag & Project Tick"
